pkg/activator/net/revision_backends.go

```diff
@@ -251,17 +255,29 @@ func (rw *revisionWatcher) checkDests(dests dests) {
 	// If we have discovered that this revision cannot be probed directly
 	// do not spend time trying.
 	if rw.podsAddressable {
+		// reprobe set contains the targets that moved from ready to non-ready set.
+		// so they have to be re-probed.
+		reprobe := curDests.becameNonReady(prevDests)
+		if len(reprobe) > 0 {
+			rw.logger.Infof("Need to reprobe pods who became non-ready: %+v", reprobe)
+			// Trim the pods that migrated to the non-ready set from the
+			// ready set from the healthy pods. They will automatically
+			// probed below.
+			rw.healthyPods = rw.healthyPods.Difference(reprobe)
```

Comment:
We don't need immutability here, so let's just iterate over the `reprobe` pods and remove them from healthyPods' map to save some allocation?

Reply:
Done.